### PR TITLE
File upload - frontend

### PR DESCRIPTION
**Pull request by DaveOrDead**
A lot of files changed in this PR - my bad, got carried away with cleanup.

The main 2 files that need reviewing are:
- `pet-details-form.php`
- `main.js`

### CODE REVIEW

File: app/partials/pet-details-form.php [hunk 1]
```diff
@@ -37,6 +39,30 @@
...
 
+    <?php if ($page != 'index') { ?>
+    <div class="h-spacing">
+        <p>Photo: </p>
+
+        <div class="c-file-upload">
+
+            <img id="pet_photo" src="" alt="Pet image" class="c-file-upload__image h-hide">
+
+            <input id="pet_photo_upload" class="c-file-upload__input h-hide-visually" type="file", name="pet_photo_upload">
```
Comment (gary-wood) [inline app/partials/pet-details-form.php]: Erroneous comma after the `type="file"` ?

File: app/src/js/main.js [hunk 2]
```diff
@@ -78,4 +78,35 @@ petDetectrApp.toggleFieldVisibility = function toggleFieldVisibility(e) {
...
+        } else {
+            imagePreview.src = '';
+        }
+
+        imagePreview.classList.remove('h-hide');
+    });
+};
+
 petDetectrApp.formSetup();
+petDetectrApp.fileUpload();
```
Comment (gary-wood) [inline app/src/js/main.js:111]: Not needed for this PR, but should we start wrapping these inside an initialiser?

File: app/partials/pet-details-form.php (hunk 1 above)
Comment (DaveOrDead) [inline app/partials/pet-details-form.php]: Good spot :) removed

File: app/src/js/main.js (hunk 2 above)
Comment (DaveOrDead) [inline app/src/js/main.js:111]: Yeah we should also start breaking these out into modules too. 

And start writing some tests

**Comment by gary-wood** [comment]
LGTM!

**Pull request merged**